crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,152 @@
+use super::{endpoint::Endpoint, octez_node::OctezNode};
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
+
+pub struct OctezClientConfig {
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    endpoint: OctezNodeEndpoint,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientConfig {
+    pub fn new(endpoint: OctezNodeEndpoint) -> Self {
+        Self {
+            binary_path: None,
+            base_dir: None,
+            endpoint,
+            disable_unsafe_disclaimer: false,
+        }
+    }
+    pub fn set_path(mut self, path: PathBuf) -> Self {
+        self.binary_path = Some(path);
+        self
+    }
+    pub fn set_base_dir(mut self, base_dir: PathBuf) -> Self {
+        self.base_dir = Some(base_dir);
+        self
+    }
+    pub fn set_endpoint(mut self, endpoint: OctezNodeEndpoint) -> Self {
+        self.endpoint = endpoint;
+        self
+    }
+    pub fn set_disable_unsafe_disclaimer(
+        mut self,
+        disable_unsafe_disclaimer: bool,
+    ) -> Self {
+        self.disable_unsafe_disclaimer = disable_unsafe_disclaimer;
+        self
+    }
+    pub fn build(mut self) -> OctezClient {
+        let base_dir = self
```

Comment:
Just curious, why not use
```rs
self.base_dir.unwrap_or( <call tempdir and return path> )
```
but create another enum `OctezClientDirectory`?

Reply:
It's fine that the directory gets removed. The client binary will create one if the path does not point to an existing directory.

Reply:
oh but don't we still want the tmp file to be deleted after the octez client goes out of scope or is that handled by the binary as well?

Reply:
```
./octez-client --base-dir ~/Desktop/oc gen keys leo            1 ↵ ──(Wed,Sep25)─┘
```
Error:
  Specified --base-dir does not exist. Please create the directory and try again.
  
i get error for gen keys still with non existent path

Reply:
I see. I think it's okay to leave the directory there because it's created in `/tmp` and the operating system will clean it up eventually. It's fine if you want to do it here. I didn't do that with `OctezNode` Maybe I should implement that later on.

Reply:
Oh I didn't know that it's different in `octez-client`. In octez-node it's fine with a non-existent `data-dir` parameter. I guess here we need to hold that directory.